lib/conversation/replies.js

```diff
@@ -33,78 +33,84 @@ function customMsg(fn) {
+ * version of the message given a replyTemplate string and args object as arguments.
  * It sends an error response to the user.
  *
  * @param  {function} fn function to be wrapped.
  * @return {function}    wrapped function
  */
 function renderEndConversationWithErrorMsg(fn) {
-  return (messageTemplate, args) => helpers.renderMessageTemplate(args.req, messageTemplate)
-      .then((replyMessage) => {
-        args.replyText = replyMessage;
+  return (replyTemplate, args) => helpers.renderMessageTemplate(args.req, replyTemplate)
+    .then((replyText) => {
+      if (!args.req.user) {
+        return helpers.sendResponse(args.res, 200, replyText, replyTemplate);
+      }
 
-        args.req.user.updateConversation(messageTemplate)
-          .then(() => {
-            fn(args);
-            stathat.postStat(`campaignbot:${messageTemplate}`);
-            BotRequest.log(args.req, 'campaignbot', messageTemplate, args.replyText);
-            newrelic.addCustomParameters({ blinkSuppressRetry: true });
-            args.res.setHeader('x-blink-retry-suppress', true);
-            return helpers.sendErrorResponse(args.res, args.error);
-          });
-      })
-      .catch(err => helpers.sendErrorResponse(args.res, err));
+      args.replyText = replyText;
+      fn(args);
+      stathat.postStat(`campaignbot:${replyTemplate}`);
+      BotRequest.log(args.req, 'campaignbot', replyTemplate, replyText);
+      newrelic.addCustomParameters({ blinkSuppressRetry: true });
+      args.res.setHeader('x-blink-retry-suppress', true);
+
+      return helpers.sendErrorResponse(args.res, args.error);
+    })
+    .catch(err => helpers.sendErrorResponse(args.res, err));
 }
 
 
 /**
  * Decorator.
  * Returns passed function wrapped with functionality to retrieve and inject a rendered
- * version of the message given a messageTemplate string and args object as arguments.
+ * version of the message given a replyTemplate string and args object as arguments.
  * It also saves the campaignId as the user's current_campaign
  * It sends a success response to the user.
  *
  * @param  {function} fn function to be wrapped.
  * @return {function}    wrapped function
  */
 function renderContinueConversationMsg(fn) {
-  return (messageTemplate, args) => helpers.renderMessageTemplate(args.req, messageTemplate)
-      .then((replyMessage) => {
-        args.replyText = replyMessage;
+  return (replyTemplate, args) => helpers.renderMessageTemplate(args.req, replyTemplate)
+      .then((replyText) => {
+        if (!args.req.user) {
+          return helpers.sendResponse(args.res, 200, replyText, replyTemplate);
+        }
 
+        args.replyText = replyText;
         // User's Current Campaign may need to be updated, pass req.campaignId to check.
-        args.req.user.updateConversation(messageTemplate, args.req.campaignId)
+        return args.req.user.updateConversation(replyTemplate, args.req.campaignId)
```

Comment:
I'm wondering why we are updating the Conversation here, but not [above](https://github.com/DoSomething/gambit/pull/965/files#diff-d00b1362d15f78f5fccb7ba59af8dffeL71).

Reply:
One of the next tasks defined in this PR description is to refactor `updateConversation` as just `updateCampaign`. We shouldn't need to update the User's `current_campaign` if the User's being directed into Agent View / Support.

The `updateConversation` function was introduced in https://github.com/DoSomething/gambit/pull/934, when we were looking to move all Gambit Conversations logic into this repo, monolith style. We don't have a need to store or update a User's `lastOutboundTemplate` in this repo anymore (especially since we'll be in a place to remove the User model completely once Conversations goes live).